cfme/tests/infrastructure/test_instance_analysis.py

```diff
@@ -0,0 +1,299 @@
+                              handle_exception=True)
+
+    # Check that we can at least get the uptime via ssh this should only be possible
+    # if the username and password have been set via the cloud-init script so
+    # is a valid check
+    ssh_client = ssh.SSHClient(hostname=connect_ip, username=provisioning['ci-username'],
+                               password=provisioning['ci-pass'], port=22)
+    wait_for(ssh_client.uptime, num_sec=200, handle_exception=True)
+
+    vm = VM.factory(vm_name, provider)
+    vm.ssh_client = ssh_client
+    return vm
+
+
+def wait_for_state_change_time_refresh(instance, state_change_time, timeout=600):
+    """ Waits for 'Last Analyzed' refresh
+    """
+    def _wait_for_state_refresh():
+        if current_version() > "5.5":
+            api_object.reload()
+        try:
+            return api_object.last_scan_on.replace(tzinfo=None) >= state_change_time
+        except AttributeError:
+            return False
+
+    rest_api = pytest.store.current_appliance.rest_api
+    api_object = rest_api.collections.vms.find_by(name=instance.name)[0]
+
+    wait_for(_wait_for_state_refresh, num_sec=timeout, delay=30)
+
+
+@pytest.mark.long_running
+def test_ssa_vm(provider, testing_instance, soft_assert):
+    """ Tests SSA can be performed and returns sane results
+
+    Metadata:
+        test_flag: vm_analysis
+    """
+
+    users_number = None
+    group_number = None
+    packages_number = None
+#    init_processes_number = None
+    system_release = testing_instance.ssh_client.run_command("cat /etc/system-release").output
+
+    if 'Red Hat Enterprise Linux' or 'Fedora' in system_release:
+        users_number = testing_instance.ssh_client.run_command(
+            "cat /etc/passwd | wc -l").output.strip('\n')
+        group_number = testing_instance.ssh_client.run_command(
+            "cat /etc/group | wc -l").output.strip('\n')
+        packages_number = testing_instance.ssh_client.run_command(
+            "rpm -qa | wc -l").output.strip('\n')
+#        init_processes_number = testing_instance.ssh_client.run_command(
+#            "chkconfig --list | wc -l").output.strip('\n')
+
+    state_change_time = datetime.utcnow()
+    testing_instance.smartstate_scan()
+    wait_for_state_change_time_refresh(testing_instance, state_change_time)
+
+    # Check that all data has been fetched
+    soft_assert(testing_instance.get_detail(properties=('Security', 'Users')) == users_number)
+    soft_assert(testing_instance.get_detail(properties=('Security', 'Groups')) == group_number)
+
+    soft_assert(testing_instance.get_detail(properties=('Configuration', 'Packages')) ==
+                packages_number)
+#    soft_assert(testing_instance.get_detail(properties=('Configuration', 'Init Processes')) ==
+#                init_processes_number)
+
+
+@pytest.mark.long_running
+def test_ssa_users(provider, testing_instance, soft_assert):
```

Comment:
We could DRY the two tests below, but probably not essential. ie.....call
```check_object('userdel {0} || useradd {0}', '/etc/group', 'Users')``` but....just a line saver really.